apps/social/src/hooks/api.ts

```diff
@@ -0,0 +1,194 @@
+          const state = await api.programState.read(program, buffer);
+
+          return state.toHuman();
+        });
+
+      const messagaArr = await Promise.all(promises);
+      const msg = messagaArr.flat() as Message[];
+      const sorted = msg.sort(
+        (prev, next) =>
+          parseInt(prev.timestamp.replaceAll(',', ''), 10) - parseInt(next.timestamp.replaceAll(',', ''), 10),
+      );
+      setMessages(sorted.reverse());
+    };
+
+    if (channels) getMessges();
+  }, [channels, api.programState]);
+
+  return messages;
+}
+
+function useOwnFeed() {
+  const [messages, setMessages] = useState<Message[] | null>();
+  const { subscriptions } = useSubscriptions();
+  const { api } = useApi();
+
+  useEffect(() => {
+    const getMessges = async () => {
+      const apiRequest = new ServerRPCRequestService();
+
+      const batchParams = subscriptions!.map((id) => ({
+        programId: id,
+        genesis: localStorage.getItem(LOCAL_STORAGE.GENESIS),
+      }));
+
+      const response = await apiRequest.getResource('program.meta.get', batchParams);
+
+      const promises = response
+        .filter((res: any) => res.result)
+        .map(async ({ result: { program, metaFile } }: any) => {
+          const buffer = Buffer.from(metaFile, 'base64');
+          const state = await api.programState.read(program, buffer);
+
+          return state.toHuman();
+        });
+
+      const messagaArr = await Promise.all(promises);
+      const msg = messagaArr.flat() as Message[];
+      const sorted = msg.sort(
+        (prev, next) =>
+          parseInt(prev.timestamp.replaceAll(',', ''), 10) - parseInt(next.timestamp.replaceAll(',', ''), 10),
+      );
+      setMessages(sorted.reverse());
+    };
+
+    if (subscriptions) getMessges();
+  }, [subscriptions, api.programState]);
+
+  return messages;
+}
+
+function useChannelActions() {
+  const apiRequest = new ServerRPCRequestService();
+  const [metadata, setMetadata] = useState<Metadata>();
+
+  const { id } = useParams() as Params;
+  const genesis = localStorage.getItem(LOCAL_STORAGE.GENESIS);
+
+  useEffect(() => {
+    apiRequest
+      .getResource('program.meta.get', [{ programId: id, genesis }])
+      .then(
+        ([
+          {
+            result: { metaFile },
+          },
+        ]) => Buffer.from(metaFile, 'base64'),
+      )
+      .then((buffer) => getWasmMetadata(buffer))
+      .then((m) => setMetadata(m));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [id]);
+
+  const sendMessage = useSendMessage(id, metadata);
+
+  const post = (text: string, onSuccess?: () => void) => {
+    const payload = { Post: text };
+    sendMessage(payload, { onSuccess });
+  };
+
+  const subscribe = (onSuccess?: () => void) => {
+    const payload = { Subscribe: 'Null' };
+    sendMessage(payload, { onSuccess });
+  };
+
+  const unsubscribe = (onSuccess?: () => void) => {
+    const payload = { Unsubscribe: 'Null' };
```

Comment:
```suggestion
    const payload = { Subscribe: null };
    sendMessage(payload, { onSuccess });
  };

  const unsubscribe = (onSuccess?: () => void) => {
    const payload = { Unsubscribe: null };
```
no need to pass `null` as a string?

Reply:
it's the same, but I fixed it for the overall beauty